node/building_blocks.re

```diff
@@ -59,30 +59,45 @@ let is_current_producer = (state, ~key) => {
 // TODO: check if block must have published a new snapshot
 let is_signable = (state, block) => {
   open Protocol.Operation.Side_chain;
-  let trusted_validators = state.Node.trusted_validator_membership_change;
+  let Node.{
+        trusted_validator_membership_change,
+        protocol: {last_seen_membership_change_timestamp, _},
+        _
+      } = state;
+  let allowed_op_reception_delay = 0.200; // Assumed max network latency
+  let current_time = Unix.time();
+  let allowed_op_window = current_time +. allowed_op_reception_delay;
+  let next_allowed_membership_change_timestamp =
+    last_seen_membership_change_timestamp +. 24. *. 60. *. 60.;
   let rec contains_only_trusted_add_validator_op = result_so_far =>
     fun
     | [] => result_so_far
     | [h, ...t] =>
       switch (h.kind) {
-      | Add_validator(validator) =>
+      | Add_validator(validator, timestamp) =>
         if (Trusted_validators_membership_change.(
               Set.mem(
                 {address: validator.address, action: Add},
-                trusted_validators,
+                trusted_validator_membership_change,
               )
-            )) {
+            )
+            && timestamp > current_time
```

Comment:
Should this be `>=`? Like when a node cli sends a Add/Remove operation to itself